### ISSUE TITLE
unittestt.SkipTest is treated as failure in setUpClass() by testtools.run

### ISSUE THREAD

**Issue by mtreinish** [testing-cabal/testtools#272]
When running a test suite built with stdlib unittest (running with Python 3.6.5) if a setUpClass() method raises a unittest.SkipTest exception to skip the test the testtools.run runner is treating that as an error. For example using a synthetic example like:

```
import unittest

class TestDirectSkip(unittest.TestCase):

    @classmethod
    def setUpClass(cls):
        super(TestDirectSkip, cls).setUpClass()
        cls.test = 'a'
        raise unittest.SkipTest("I really skip")

    def test_foo(self):
        self.assertEqual(self.test, 'a')
```
will return a failure like:

```
Tests running...
======================================================================
ERROR: setUpClass (test_skips.TestDirectSkip)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/tests/test_skips.py", line 29, in setUpClass
    raise unittest.case.SkipTest("I really skip")
unittest.case.SkipTest: I really skip
```

My thinking it it has something to do with: https://github.com/testing-cabal/testtools/blob/master/testtools/testcase.py#L66-L69 because if you switch the exception from unittest to unittest2 in the example like:

```
import unittest

class TestDirectSkip(unittest.TestCase):

    @classmethod
    def setUpClass(cls):
        super(TestDirectSkip, cls).setUpClass()
        cls.test = 'a'
        import unittest2
        raise unittest2.SkipTest("I really skip")

    def test_foo(self):
        self.assertEqual(self.test, 'a')
```
The skip is properly recognized.

**Comment by rbtcollins** [comment]
Interesting. Can you check the actual exception types at play here?

**Comment by mtreinish** [comment]
The reverse case is also an issue here. The `unittest2.case.SkipTest` raised by testtools for `skipException` is treated as a failure by stdlib unittest. For example, running:

```
import testtools

class TestDirectSkip(testtools.TestCase):

    @classmethod
    def setUpClass(cls):
        super(TestDirectSkip, cls).setUpClass()
        cls.test = 'a'
        raise self.skipException("I really skip")

    def test_foo(self):
        self.assertEqual(self.test, 'a')
```

with `python -m testtools.run` will skip as expected. But when you run it with `python -m unittest` you
get a failure like:

```
E
======================================================================
ERROR: setUpClass (test_skip.TestDirectSkip)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/test_skip.py", line 9, in setUpClass
    raise cls.skipException("I really skip")
unittest2.case.SkipTest: I really skip

----------------------------------------------------------------------
Ran 0 tests in 0.000s

FAILED (errors=1)
```

**Comment by stmcginnis** [comment]
I got around this by changing https://github.com/testing-cabal/testtools/blob/2d05afacd4cc9b8f5d0af14599c971895d8ef568/testtools/testcase.py#L70-L73 to:

```python
# Handle raising SkipTest exceptions for both unittest and unittest2
_SkipTest = try_import('unittest.case.SkipTest')
_SkipTest2 = try_import('unittest2.case.SkipTest', object)

class TestSkipped(_SkipTest, _SkipTest2):
    """Raised within TestCase.run() when a test is skipped."""
```

I would submit it, but I can't even get unit tests to pass, so I'm actually a little more concerned about other issues with the package right now.

**Comment by cdent** [comment]
Is anyone maintaining a healthy fork of testtools anywhere? Seems like it may be needed?

**Comment by mtreinish** [comment]
This has been fixed by #285 and released in 2.5.0
